### DISCUSSION TITLE
On-screen clock

### DISCUSSION THREAD

**Discussion by therentabrain** [roleoroleo/yi-hack-Allwinner-v2#586] [Q&A]
Hi!  Thank you for this amazingly useful utility!  I just got a Home 1080p RFUS with the latest avail firmware and it worked perfectly right away.

Is there a way to remove the on-screen clock at the top left?  I assume that's baked into the firmware, but I thought I'd ask.

I'm using it with OBS, and so far so good, though there's a bit of lag.  Any tips for using these with OBS (esp reducing lag) I'd love it!  thank you.  I figure I'll just crop the top of the image to avoid the timestamp.

**Comment by roleoroleo** [comment]
Sorry buy I can't remove the osd (at the moment).
I didn't find a setting to disable it.

**Reply by roleoroleo** [reply]
Maybe I found an experimental solution.
Do you want to try?
Run this command and check your current status:
`hexdump -C -n 16 /tmp/mmap.info`
Then run this command and check the stream:
`printf "\03\00\00\00" | dd of=/tmp/mmap.info bs=1 seek=8 count=4 conv=notrunc`

To restore default restart the cam.

**Reply by roleoroleo** [reply]
What's the suffix of your cam?

**Reply by therentabrain** [reply]
I'm a long time dabbler in many programming languages, but I'm not sure how to answer :)
When you say run the command, is that in the web interface?  Or do I need to like ssh in or something?
I haven't done that in ages but I am sure I can.

Model suffix in the web interface shows y291ga

Thank you for looking into this!

For the record I may have found a solution to the latency problem.  I'm planning on trying it in the next few days and if it works, I'll post about it.

**Reply by therentabrain** [reply]
Also is there a way to turn the camera on and off to save power, from Windows?  Maybe via HTTP?  I hate to leave things turned on when I'm not using them, but I'd also rather not have to remember to plug and unplug it whenever I'm streaming.  I am not sure how much power it would really save, but I want to do the right thing...

I hope it's ok to hijack my own thread with multiple questions :)

**Reply by roleoroleo** [reply]
> I'm a long time dabbler in many programming languages, but I'm not sure how to answer :) When you say run the command, is that in the web interface? Or do I need to like ssh in or something? I haven't done that in ages but I am sure I can.

You should enter with a ssh session (like putty) and run the command.

> Also is there a way to turn the camera on and off to save power, from Windows? Maybe via HTTP?

Yes, it's possible to turn on and off the cam via HTTP or via MQTT, but I think this thing will not save you energy.
Turning on and off simply switches the stream: no video and no images.
Check the wiki:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Web-services-description#camera-settings

**Comment by therentabrain** [comment]
```BusyBox v1.27.2 () built-in shell (ash)


            __    __   _
            \ \  / /  | |
             \ \/ /   | |
              \  /    | |
              / /     | |
             /_/      |_|
 ------------------------------------
 Welcome to XiaoYi Linux.
 ------------------------------------

root@yi-7a2d:~# hexdump -C -n 16 /tmp/mmap.info
00000000  11 00 00 00 11 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~# printf "\03\00\00\00" | dd of=/tmp/mmap.info bs=1 seek=8 count=4 conv=notrunc
4+0 records in
4+0 records out
4 bytes (4B) copied, 0.000251 seconds, 15.6KB/s
root@yi-7a2d:~# hexdump -C -n 16 /tmp/mmap.info
00000000  11 00 00 00 11 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~#
```

Been watching the RTSP the whole time and the OSD didn't change.  

I'm using the low bandwith RTSP right now because it seems to be the tiniest bit less latency, and the image is in a small part of the main window so it doesn't matter.  But let me know if that changes anything! ... Ok I just switched to the high one.

```root@yi-7a2d:~#  hexdump -C -n 16 /tmp/mmap.info
00000000  11 00 00 00 11 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~# printf "\03\00\00\00" | dd of=/tmp/mmap.info bs=1 seek=8 count=4 conv=notrunc
4+0 records in
4+0 records out
4 bytes (4B) copied, 0.000256 seconds, 15.3KB/s
root@yi-7a2d:~#  hexdump -C -n 16 /tmp/mmap.info
00000000  11 00 00 00 11 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~#
```

Looks the same to me and the clock is there.

I think when I toggled off the IR LED, the status LED turned off.  
The option for Status LED is already off.
When I turn Status LED on, it doesn't seem to change anything as far as I can tell.

**Reply by roleoroleo** [reply]
> Looks the same to me and the clock is there.

Try these:

Dump current status:
`# hexdump -C -n 16 /tmp/mmap.info`
Enable:
`# printf "\01\00\00\00\x10\00\00\00\00\00\00\00" | dd of=/tmp/mmap.info bs=1 count=12 conv=notrunc`
Disable:
`# printf "\x10\00\00\00\x10\00\00\00\03\00\00\00" | dd of=/tmp/mmap.info bs=1 count=12 conv=notrunc`

> When I turn Status LED on, it doesn't seem to change anything as far as I can tell.

Did you try to restart the cam? I don't know if a reboot is needed with the last fw version.

**Reply by therentabrain** [reply]
> Did you try to restart the cam? I don't know if a reboot is needed with the last fw version.

I did, a few times since then.  No worries, this bit isn't a big deal, but if I can help I'm here for it.

I tried the new idea and it didn't affect the OSD.  Do I need to restart something in order for it to take effect?

```
root@yi-7a2d:~# hexdump -C -n 16 /tmp/mmap.info
00000000  11 00 00 00 11 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~# printf "\01\00\00\00\x10\00\00\00\00\00\00\00" | dd of=/tmp/mmap.info bs=1 count=12 conv=notrunc
12+0 records in
12+0 records out
12 bytes (12B) copied, 0.000295 seconds, 39.7KB/s
root@yi-7a2d:~# printf "\x10\00\00\00\x10\00\00\00\03\00\00\00" | dd of=/tmp/mmap.info bs=1 count=12 conv=notrunc
12+0 records in
12+0 records out
12 bytes (12B) copied, 0.003376 seconds, 3.5KB/s
root@yi-7a2d:~# hexdump -C -n 16 /tmp/mmap.info
00000000  10 00 00 00 10 00 00 00  03 00 00 00 68 74 74 70  |............http|
00000010
root@yi-7a2d:~#
```

**Reply by roleoroleo** [reply]
Strange.
Disable command should stop the timestamp.
Mmmmmmh.
I will check again.

**Comment by therentabrain** [comment]
I will try it soon!  thank you so much

**Comment by uzior** [comment]
Hi!

Nice thread!
I have a very similar problem, but the opposite. I would like an OSD watermark with date and time on the recordings.
 Is it possible to enable this? y21ga

**Reply by roleoroleo** [reply]
Yes it's possible, check the last commits:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/commit/4133cae56025b2f04987bc38ff5b28e0a2775a40
https://github.com/roleoroleo/yi-hack-Allwinner-v2/commit/c3e14614b9f46e3026daa66413e2a721a3b2da1f

**Reply by therentabrain** [reply]
Oh cool do we think it will work for me now?

Beginner question (I checked the wiki/faq): how best to update the yi-hack?  Download the whole hack and unzip it again into the memory card?  Download these two changed files and replace the ones that are there?  Or is there a better way?

**Reply by roleoroleo** [reply]
> Oh cool do we think it will work for me now?

I don't know. I found the command to activate it but I don't know if it works to de-activate it.

At the moment, this commit is not released. I need to rebuild the hack.
Do you want a beta to test?

**Reply by therentabrain** [reply]
You can give me one if you like; I'm busy until Thursday probably but I'd try more things!

**Comment by roleoroleo** [comment]
[y21ga_0.2.6.tgz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10911472/y21ga_0.2.6.tgz)
[y291ga_0.2.6.tgz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10911474/y291ga_0.2.6.tgz)

**Reply by therentabrain** [reply]
Ok I tried y291ga.  I didn't export my settings first, though I did back up the whole card contents.  
It booted up with the OSD on.  I looked in the menu; the OSD was supposed to be off.
I tried turning OSD on with the pref switch and no change.  Rebooted camera, no change.  Turned it back off, no change; rebooted, no change.

**Reply by therentabrain** [reply]
I read more into it and thought I'd try this.  None of these had any effect, though I wasn't restarting the camera:

```
root@yi-7a2d:~# set_tz_offset -c osd -o off
root@yi-7a2d:~# set_tz_offset -c osd -o off -f 12.0.35.8_202211241029
root@yi-7a2d:~# set_tz_offset -c osd -o off -f 12.0.35.8_202211241029 -m y291ga
root@yi-7a2d:~# set_tz_offset -v 10
root@yi-7a2d:~# set_tz_offset -v 3
```
No responses, then I realized I hadn't understood so I did:
```
root@yi-7a2d:~# set_tz_offset -c tz_offset -v 5 -f 12.0.35.8_202211241029 -m y291ga
Setting timezone to 5000 ms
```
This did work and changed the time shown.  I assume there's no reason to include all those args but it worked, so, sharing.

Let me know how I can help some more.

If we kill the clock, will the little square logo watermark at the bottom left go too?  I hardly even thought about it.

**Reply by therentabrain** [reply]
p.s. I see the bit about cloud and payload.  I have cut this camera off from the internet - would that have something to do with it?

**Reply by roleoroleo** [reply]
This command:
`set_tz_offset -c osd -o off`
should disable osd.
But it should be run after dispatch and before rmm.

This command:
`set_tz_offset -c osd -o on`
should enable osd and you can run it when you want.

This command:
`set_tz_offset -c tz_offset_osd -m y291ga -f 12 -v 3600`
set the timezone of the osd (CET in this case).

If the cam doesn't show the osd natively, the above procedure works.
I don't know if it works with your model that shows the osd natively. I can't test it.

Does the osd stops counting when you disable it (1st command)?

The logo in the lower-left can't be removed.
You can change it with the image you want but you can't remove it, because the software doesn't support alpha channel.

> p.s. I see the bit about cloud and payload. I have cut this camera off from the internet - would that have something to do with it?

This is not a problem.

**Reply by therentabrain** [reply]
> If the cam doesn't show the osd natively, the above procedure works. I don't know if it works with your model that shows the osd natively. I can't test it.
> Does the osd stops counting when you disable it (1st command)?

Nope, those OSD commands have no effect that I can notice.

> The logo in the lower-left can't be removed. You can change it with the image you want but you can't remove it, because the software doesn't support alpha channel.

Can I change it to one black pixel?

Cropping the edge has been no problem and the camera stays on forever pretty well so I'm still fine how it is :)

**Reply by roleoroleo** [reply]
> Can I change it to one black pixel?

Yes, it's possible but not simple.
You should substitute the file /home/app/main_kami.bmp (for high res) and sub_kami.bmp (for low res) with your custom image, before to load rmm. For example with a mount bind.
The image must be a specific format.
See attached archive.
[main_kami.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10954842/main_kami.zip)

**Reply by SpectralGM** [reply]
> This command: `set_tz_offset -c osd -o off` should disable osd. But it should be run after dispatch and before rmm.
> 
> This command: `set_tz_offset -c osd -o on` should enable osd and you can run it when you want.
>
Hello there! Thanks for all your hard work on this firmware.

I have a RFUS/y211ga running yi-hack-AllWinner-v2: 0.2.6. 

I have it streaming a low-res RTSP stream to my MotionEye docker container, so I don't need the built in OSD, and would prefer to use the one from my NVR, but I can't figure out a way to disable the OSD. I came across this thread, but I don't think I have these command options in my firmware. Is there a special build I can test out for you?

**Reply by roleoroleo** [reply]
It's a beta build. If you want to try it I have to prepare a new file for your model.

**Reply by SpectralGM** [reply]
I'm willing to try one out if you are willing to prep the file, but don't go out of your way on behalf of me.

**Reply by roleoroleo** [reply]
Here it is:
[y211ga_0.2.6.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10984100/y211ga_0.2.6.tar.gz)
